gitlabform/processors/multiple_entities_processor.py

```diff
@@ -16,22 +16,52 @@ def __init__(
         self,
         configuration_name: str,
         gitlab: GitLab,
-        list_method_name: str,
-        add_method_name: str,
-        delete_method_name: str,
+        # TODO: Should the option of passing as str be deprecated ? It makes life harder (for Mypy and for the devs),
+        #    and it's not like the method is not known|doesn't exist at compile-time
+        list_method_name: Union[str, Callable[[str], list]],
+        add_method_name: Union[str, Callable[[str, Any], None]],
+        delete_method_name: Union[str, Callable[[str, dict], None]],
         defining: AbstractKey,
         required_to_create_or_update: AbstractKey,
-        edit_method_name=None,
+        edit_method_name: Union[str, Callable[[str, str, Any], None], None] = None,
     ):
+        """
+
+        :param configuration_name: The key in the YAML cfg file that determines the start of this Entity's section
+        :param gitlab:
+        :param list_method_name:
+        :param add_method_name:
+        :param delete_method_name:
+        :param defining: TODO -> what is this for ?
+        :param required_to_create_or_update: Entries which (according to the API documentation) are required fields
+        :param edit_method_name: TODO -> how does it handle the actual and new cfg diff ?
```

Comment:
The `edit_method_name` method is just called with the whole config of a given entity from the config. We don't do anything smart here.